src/quo/components/wallet/approval_label/view.cljs

```diff
@@ -0,0 +1,67 @@
+(ns quo.components.wallet.approval-label.view
+  (:require [oops.core :refer [oget]]
+            [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icon]
+            [quo.components.markdown.text :as text]
+            [quo.components.wallet.approval-label.schema :as approval-label.schema]
+            [quo.components.wallet.approval-label.style :as style]
+            [quo.foundations.colors :as colors]
+            quo.theme
+            [react-native.core :as rn]
+            [react-native.hole-view :as hole-view]
+            [schema.core :as schema]
+            [utils.i18n :as i18n]))
+
+(def ^:private status-icons
+  {:approve   :i/alert
+   :approving :i/pending-state
+   :approved  :i/check})
+
+(def ^:private status-message
+  {:approve   :t/approve-amount-symbol
+   :approving :t/approving-amount-symbol
+   :approved  :t/approved-amount-symbol})
+
+(defn- view-internal
+  [{:keys [status customization-color token-value token-symbol
+           container-style button-props]
+    :or   {customization-color :blue}}]
+  (let [theme                                 (quo.theme/use-theme)
+        [container-width set-container-width] (rn/use-state 0)
+        on-layout                             (rn/use-callback #(set-container-width
+                                                                 (oget % :nativeEvent :layout :width)))]
+    [hole-view/hole-view
+     {:holes     [{:x                       0
+                   :y                       0
+                   :height                  24
+                   :width                   container-width
+                   :borderBottomStartRadius 16
+                   :borderBottomEndRadius   16}]
```

Comment:
Can we use kebab-case here?

Reply:
I think we need to set hole view up properly, or else it's just weird 👽

Reply:
As far as I understand, Reagent doesn't automatically convert kebab-case to camelCase for nested maps. Therefore, we need to use camelCase here. I'm not sure about the best way to fix this.

cc: @ilmotta

Reply:
@ajayesivan have you tried using it with kebab-case? I believe it should work
```
    [hole-view/hole-view
     {:holes     [{:x                       0
                   :y                       0
                   :height                  style/top-hole-view-height
                   :width                   container-width
                   :border-bottom-start-radius 16
                   :border-bottom-end-radius   16}]
```

Reply:
Yes. I tried and it doesn't work.

Reply:
Oh I see.. I am wondering how does kebab-case work for other components like `blur` where the definition is similar:
```
(def hole-view (reagent/adapt-react-class RNHoleView))
(def view (reagent/adapt-react-class (.-BlurView blur)))
```

Reply:
Oh that's right.. we are trying to change the case of a nested value not the prop key

Reply:
> it could be that the solution provided by a custom Reagent compiler would perform just as well as the current solution now (btw there's a proper Reagent compiler protocol).

Good to know!

> I think it's more a matter of effort than anything else. Given our priorities and that hole props don't appear very often in the codebase, it doesn't make much sense to invest our time on such improvements.

@ilmotta that's right, I agree!

Reply:
@ilmotta  @OmarBasem  @ajayesivan 

Just a side comment.

This problem would also be solved if we explore the idea I mentioned in:
- https://github.com/status-im/status-mobile/issues/20110#issuecomment-2120881758